### PR TITLE
[CHK-2262] update api npg client

### PR DESCRIPTION
**Pull request by a-mannia**
#### List of Changes
Updated API key header management for NPG APIs.

#### Motivation and Context
The management of the API keys for the NpgClient exposes potential side effects when concurrent access to the client occurs. With this change, the API key is specified for each API call, rather than once when the client is instantiated.

#### How Has This Been Tested?
Unit Test

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java [hunk 1]
```diff
@@ -513,10 +501,7 @@ public Mono<StateResponseDto> confirmPayment(
                         )
                 )
                         .onErrorMap(err -> exceptionToNpgResponseException(err, span, gatewayOperation)),
-                span -> {
-                    paymentServicesApi.getApiClient().setApiKey(null);
-                    span.end();
-                }
+                span -> span.end()
```
Comment (petretiandrea) [inline src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java:504]: Just a little fix
```suggestion
                Span::end
```

**Review by petretiandrea — CHANGES_REQUESTED** [review]

**Review by pietro-tota — DISMISSED** [review]
Well done, just take into account the suggestion :)

File: src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java (hunk 1 above)
Comment (a-mannia) [reply src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java]: Done.

**Review by petretiandrea — APPROVED** [review]
Well done 🚀

**Review by ciuffagianluca — APPROVED** [review]

**Review by infantesimone — APPROVED** [review]

**Pull request merged**